pkg/dvo/builder.go

```diff
@@ -0,0 +1,56 @@
+package dvo
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/openshift/managed-upgrade-operator/pkg/metrics"
+	"github.com/openshift/managed-upgrade-operator/util"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// DvoClientBuilder enables implementation of a DVO client.
+//go:generate mockgen -destination=mocks/builder.go -package=mocks github.com/openshift/managed-upgrade-operator/pkg/dvo DvoClientBuilder
+type DvoClientBuilder interface {
+	New(c client.Client) (DvoClient, error)
+}
+
+// NewBuilder creates a new DvoClientBuilder instance
+func NewBuilder() DvoClientBuilder {
+	return &dvoClientBuilder{}
+}
+
+type dvoClientBuilder struct{}
+
+// New creates a new instance of DvoClient.
+// It takes a client.Client as input and returns a DvoClient interface and an error.
+func (dcb *dvoClientBuilder) New(c client.Client) (DvoClient, error) {
+
+	// Get the service URL for the deployment-validation-operator-metrics service
+	svcURL, err := metrics.NetworkTarget(c, "openshift-deployment-validation-operator", "deployment-validation-operator-metrics", "8383")
+	if err != nil {
+		return nil, err
+	}
+
+	// Print the service URL for debugging purposes
+	fmt.Printf("********%s", svcURL)
```

Comment:
Will we keep this in future? Or it will be removed before merge

Reply:
yeah I will make sure to remove the testing loggers before merge. Thank you